client-tools/Dump/DumpFeature.cpp

```diff
@@ -1854,9 +1867,10 @@ void DumpFeature::ParallelDumpServer::runWriterThread() {
         FATAL_ERROR_EXIT();
       }
 
-      auto file = fileProvider->getFile(it2->second.collectionName);
-      filesByShard.emplace(shardId, file);
-      return file;
+      std::string collectionName = it2->second.collectionName;
```

Comment:
Maybe string_view? It looks like shards map destroyed not early than filesByShard

Reply:
I also thought about using an `std::string_view` initially, and I think it could be done. 
But it makes things harder to reason about. For example, the `collectionName` is handed to `DumpFileProvider::getFiles(std::string const&)`. So this method would need to change its signature.  And then this method does a lookup using that filename in an associative container (`_collectionInfo`), which is an `std::map<std::string, ...>`. So I would need to change this too.
Additionally returning an std::string_view from a function/lambda as we would do here if we change the return type of the lambda makes it harder for the caller to reason about until when the std::string_view will remain valid, and it is possible to introduce ownership problems should the code change in the future. Returning an std::string is self-explanatory and there are no ownership issues. So I prefered simplicity over performance here, which should be ok, as this is not really a hot function compared to what else is happening when dumping collections.